core/drand_beacon_control.go

```diff
@@ -801,31 +801,49 @@ func (bp *BeaconProcess) PingPong(c context.Context, in *drand.Ping) (*drand.Pon
 
 func (bp *BeaconProcess) RemoteStatus(c context.Context, in *drand.RemoteStatusRequest) (*drand.RemoteStatusResponse, error) {
 	replies := make(map[string]*drand.StatusResponse)
+	bp.log.Debugw("Starting remote status request", "for_nodes", in.GetAddresses())
 	for _, addr := range in.GetAddresses() {
-		if addr.Address == bp.priv.Public.Addr {
-			// no need to reach us
+		remoteAddress := addr.GetAddress()
+		if remoteAddress == "" {
+			bp.log.Errorw("Received empty address during remote status", "addr", addr)
 			continue
 		}
-		p := net.CreatePeer(addr.GetAddress(), addr.Tls)
-		resp, err := bp.privGateway.Status(c, p, &drand.StatusRequest{
+
+		var err error
+		var resp *drand.StatusResponse
+		statusReq := &drand.StatusRequest{
 			CheckConn: in.GetAddresses(),
-		})
+			Metadata:  bp.newMetadata(),
+		}
+		if remoteAddress == bp.priv.Public.Addr {
+			// it's ourself
+			resp, err = bp.Status(c, statusReq)
+		} else {
+			bp.log.Debugw("Sending status request", "for_node", remoteAddress, "has_TLS", addr.Tls)
+			p := net.CreatePeer(remoteAddress, addr.Tls)
+			resp, err = bp.privGateway.Status(c, p, statusReq)
+		}
 		if err != nil {
-			bp.log.Debug("Remote Status", addr, " FAIL", err)
+			bp.log.Errorw("Status request failed", "remote", addr, "error", err)
```

Comment:
so will erroring nodes just be skipped in the output status payload? It might be nice to show specific errors for them

Reply:
This would require reworking things at the protobuf layer too, let's skip it for now. It's in the nodes logs.